lib/schema.js

```diff
@@ -13,18 +13,262 @@ class Schema {
      * @param {Object} [data.raw] Raw schema
      * @param {Number} [data.time] Time when the schema was made
      */
-    constructor (data) {
+    constructor(data) {
         this.version = data.version || null;
         this.raw = data.raw || null;
         this.time = data.time || new Date().getTime();
     }
 
+    /**
+     * Gets sku
+     * @param {string} name
+     * @returns {string|null} SKU
+     */
+    getSkuFromName(name) {
+        return SKU.fromObject(this.getItemObjectFromName(name));
+    }
+    /**
+     * Gets sku item object
+     * @param {string} name
+     * @returns {object} SKU
+     */
+    getItemObjectFromName(name) {
+        name = name.toLowerCase();
+        const item = {
+            defindex: null,
+            quality: null,
+            craftable: true
+        };
+
+        if (
+            name.includes('strange part') ||
+            name.includes('strange cosmetic part') ||
+            name.includes('strange filter') ||
+            name.includes('strange count transfer tool') ||
+            name.includes('strange bacon grease')
+        ) {
+            const schemaItem = this.getItemByItemName(name);
+            if (!schemaItem) return item;
+            item.defindex = schemaItem.defindex;
+            item.quality = item.quality || schemaItem.item_quality; //default quality
+            return item;
+        }
+
+        if (name.includes('strange')) {
+            name = name.replace('strange', '').trim();
+            item.quality = 11;
+        }
+        //Get all qualities
+        const qualities = Object.keys(this.raw.schema.qualities).reduce(
+            (obj, q) => {
+                obj[this.raw.schema.qualityNames[q].toLowerCase()] = this.raw.schema.qualities[q];
+                return obj;
+            },
+            {}
+        );
+        //Try to find quality name in name
+        for (const quality in qualities) {
+            if (name.includes(quality)) {
+                name = name.replace(quality, '').trim();
```

Comment:
Need to add some exception here:
- "Haunted Ghosts" (Unusual Effect and a skin name)
- "Haunted Phantasm Jr"
- "Haunted Phantasm"
- "Haunted Metal Scrap"
- "Haunted Hat"
- "Vintage  Tyrolean"
- "Vintage Merryweather"
- "Unusual Cap"

Optional (maybe not need):
- "Brazil Fortress Halloween - Haunted Group Winner"
- "Brazil Fortress Halloween - Haunted Group Participant"
- "Summer 2012 Community Bundle"
- "Halloween 2012 Community Bundle"
- "Halloween 2013 Community Bundle"
- "TF2mixup Community Event 2013 Promo"
- "Halloween 2014 Community Bundle"
- "TF2mixup Community Event 2014 Promo"
- "Tough Break Community Maps Bundle"
- "Robo Community Crate"
- "Robo Community Crate Key"
- "Pile of Robo Community Crate Key Gifts"
- "A Random Robo Community Crate Key Gift"
- "End of the Line Community Crate"
- "Invasion Community Update Pass"
- "Invasion Community Update Coin"
- "Invasion Community Update Key"
- "AU Highlander Community League First Place"
- "AU Highlander Community League Second Place"
- "AU Highlander Community League Third Place"
- "Palaise MGE Doubles Community Clash Champion"
- "Palaise MGE Doubles Community Clash Runner-Up"
- "Palaise MGE Doubles Community Clash Contestant"
- "DreamHack Community Clash 1st Place"
- "DreamHack Community Clash 2nd Place"
- "DreamHack Community Clash 3rd Place"
- "DreamHack Community Clash Participant"
- "DreamHack Community Clash Contributor"
- "End of the Line Community Update Medal"
- "Invasion Community Update Medal"
- ...

Based on all of this, I think the best line of code is:
```js
            const exception = [
                'Haunted Ghosts',
                'Haunted Phantasm Jr',
                'Haunted Phantasm',
                'Haunted Metal Scrap',
                'Haunted Hat',
                'Vintage  Tyrolean',
                'Vintage Merryweather',
                'Unusual Cap'
            ];
            if (name.startsWith(quality) && !exception.some(ex => name.includes(ex))) {
                name = name.replace(quality, '').trim();
                // ...
            }
```